src/openrct2/object/Object.cpp

```diff
@@ -329,6 +329,48 @@ std::unique_ptr<IStream> ObjectAsset::GetStream() const
     return {};
 }
 
+u8string VersionString(const ObjectVersion& version)
+{
+    return std::to_string(std::get<0>(version)) + "." + std::to_string(std::get<1>(version)) + "."
+        + std::to_string(std::get<2>(version));
+}
+
+ObjectVersion VersionTuple(std::string_view version)
+{
+    if (version.empty())
+    {
+        return std::make_tuple(0, 0, 0);
+    }
+
+    auto nums = String::Split(version, ".");
+    uint16_t versions[VersionNumFields] = {};
+    if (nums.size() > VersionNumFields)
+    {
+        log_warning("%i fields found in version string '%s', expected X.Y.Z", nums.size(), version);
+    }
+    if (nums.size() == 0)
+    {
+        log_warning("No fields found in version string '%s', expected X.Y.Z", version);
+        return std::make_tuple(0, 0, 0);
+    }
```

Comment:
What about a version format like "23.12"?

Reply:
For the warning?

Reply:
Generally speaking, should it accept such version format or not because you are not checking for such case.

Reply:
Yes, it is intended to support 1 to 3 values. I didn't write that in the error because I didn't think of a good message that conveys that information succintly